grakn-core/src/main/java/ai/grakn/concept/Thing.java

```diff
@@ -80,6 +80,14 @@
     Thing resource(Resource resource);
 
     /**
+     * Creates a relationship from this instance to the provided resource.
+     *
+     * @param resource The resource to which a relationship is created
+     * @return The relationship connecting the thing and the resource
+     */
+    Relationship resourceRelationship(Resource resource);
```

Comment:
I feel like if Graql can do it, then you should be able to do it with the graph API too. Maybe there's a better way to present it though.